### PR TITLE
refactor: CDS logs

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed Changes

_[describe the changes here...]_

## Linked Issues

_[if applicable, add links to issues resolved by this PR]_

## Types of changes

_What type of change does this pull request make (put an `x` in the boxes that apply)?_

- [ ] Bug fix (non-breaking change that fixes an issue).
- [ ] New feature added (non-breaking change that adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected).
- [ ] Documentation update.
- [ ] Something else (e.g., tests, scripts, example, deployment, infrastructure).

## Checklist

_Put an `x` in the boxes that apply:_

- [ ] I have read the [CONTRIBUTING](https://github.com/fetchai/fetchd/blob/main/CONTRIBUTING.md) guide
- [ ] Checks and tests pass locally

### If applicable

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease
- [ ] I have added/updated the documentation

## Further comments

_[if this is a relatively large or complex change, kick off a discussion by explaining why you chose the solution you did, what alternatives you considered, etc...]_

### CODE REVIEW

File: app/upgrade_cudos.go [hunk 1]
```diff
@@ -767,7 +767,7 @@ func withdrawGenesisStakingDelegations(genesisData *GenesisData, cudosCfg *Cudos
 		return fmt.Errorf("remaining bonded pool balance %s is too high", bondedPool.balance.String())
 	}
 
-	println("remaining bonded pool balance: ", bondedPool.balance.String())
+	app.Logger().Info("remaining bonded pool balance: ", bondedPool.balance.String())
```
Comment (pbukva) [inline app/upgrade_cudos.go:770]: Please add `cudos merge: ` prefix for all log records like this:
```suggestion
	app.Logger().Info("cudos merge: remaining bonded pool balance: ", bondedPool.balance.String())
```

File: app/upgrade_v_11_4_manifest.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ import (
 const manifestFilenameBase = "upgrade_manifest.json"
 
 type UpgradeManifest struct {
+	ConfigSource       string                     `json:"config_source"`
```
Comment (pbukva) [inline app/upgrade_v_11_4_manifest.go:16]: Manifest must be deterministic (the same manifest data must be generated everywhere). As the implication, it must **\*not\*** contain data which are client specific, like for example file path.
All we can put here are sh256 hashes of the input files:
```suggestion
        // Following 2 hash data mambers are intentionally without `omitempty` parameter in `json:...` decorator
	GenesisFileSha256        string `json:"genesis_file_sha256"`
	NetworkConfigFileSha256  string `json:"network_config_file_sha256"`  
```

File: app/app.go [hunk 3]
```diff
@@ -727,17 +727,21 @@ func (app *App) GetSubspace(moduleName string) paramstypes.Subspace {
...
 
-func getNetworkInfo(app *App, ctx sdk.Context) (*NetworkConfig, error) {
+func getNetworkInfo(app *App, ctx sdk.Context, manifest *UpgradeManifest) (*NetworkConfig, error) {
 	// Load network config from file if given
 	var networkInfo *NetworkConfig
 	var err error
 	if app.cudosMigrationConfigPath != "" {
+		app.Logger().Info("Loading network config", "file", app.cudosMigrationConfigPath, "hash", app.cudosMigrationConfigSha256)
 		networkInfo, err = LoadNetworkConfigFromFile(app.cudosMigrationConfigPath, &app.cudosMigrationConfigSha256)
+		manifest.ConfigSource = fmt.Sprintf("file %s %s", app.cudosMigrationConfigPath, app.cudosMigrationConfigSha256)
```
Comment (pbukva) [inline app/app.go:737]: Manifest must  **\*not\***  conatin file path of input files - see [this comment](https://github.com/fetchai/fetchd/pull/383/files#r1786792275).

File: app/upgrade_v_11_4_network_config.go [hunk 4]
```diff
@@ -265,6 +268,7 @@ var NetworkInfos = map[string]NetworkConfig{
 }
 
 type NetworkConfig struct {
+	ChainID            string                `json:"chain_id"`
```
Comment (pbukva) [inline app/upgrade_v_11_4_network_config.go:271]:
```suggestion
        GenesisFileSha256       string                     `json:"genesis_file_sha256"`
	NetworkConfigFileSha256 string                     `json:"network_config_file_sha256"`
```

File: app/upgrade_cudos_distribution.go [hunk 5]
```diff
@@ -396,12 +396,12 @@ func withdrawGenesisDistributionRewards(app *App, genesisData *GenesisData, cudo
...
 	communityBalance, _ := genesisData.distributionInfo.feePool.communityPool.TruncateDecimal()
 	remainingBalance := distributionModuleAccount.balance.Sub(communityBalance)
-	app.Logger().Info("Remaining dist balance", "amount", remainingBalance.String())
+	app.Logger().Info("cudos merge: remaining dist balance", "amount", remainingBalance.String())
 
 	// TODO: Write to manifest?
 	err = checkTolerance(remainingBalance, maxToleratedRemainingDistributionBalance)
 	if err != nil {
-		return fmt.Errorf("Remaining distribution balance %s is too high", remainingBalance.String())
+		return fmt.Errorf("remaining distribution balance %s is too high", remainingBalance.String())
```
Comment (pbukva) [inline app/upgrade_cudos_distribution.go:404]:
```suggestion
		return fmt.Errorf("cudos merge: remaining distribution balance %s is too high", remainingBalance.String())
```

File: app/app.go [hunk 6]
```diff
@@ -732,17 +732,28 @@ func getNetworkInfo(app *App, ctx sdk.Context, manifest *UpgradeManifest) (*Netw
 	var networkInfo *NetworkConfig
 	var err error
 	if app.cudosMigrationConfigPath != "" {
-		app.Logger().Info("Loading network config", "file", app.cudosMigrationConfigPath, "hash", app.cudosMigrationConfigSha256)
+		app.Logger().Info("cudos merge: loading network config", "file", app.cudosMigrationConfigPath, "hash", app.cudosMigrationConfigSha256)
```
Comment (pbukva) [inline app/app.go:735]: (this comment belongs to line [730](https://github.com/fetchai/fetchd/pull/383/files/959f5dd25a6f9ef8fd5a512405ee79c400879fe0..615a29b4b604b9aecbbff8ceae440357058b5251#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7R730) - github UI didnlt allow me to put it there)
We would need to add new parameter either for the input genesis  content, or for `mergeSourceChainId string`, in order to pass in the Merge Source ChainID value in for the check:
```go
func getNetworkInfo(app *App, ctx sdk.Context, mergeSourceGenesisAppStateJson *map[string]interface{}, manifest *UpgradeManifest) (*NetworkConfig, error) {
```

File: app/app.go [hunk 7]
```diff
@@ -732,17 +732,28 @@ func getNetworkInfo(app *App, ctx sdk.Context, manifest *UpgradeManifest) (*Netw
...
 		networkInfo, err = LoadNetworkConfigFromFile(app.cudosMigrationConfigPath, &app.cudosMigrationConfigSha256)
-		manifest.ConfigSource = fmt.Sprintf("file %s %s", app.cudosMigrationConfigPath, app.cudosMigrationConfigSha256)
 		if err != nil {
 			return nil, err
 		}
+
+		if networkInfo.ChainID != ctx.ChainID() {
+			return nil, fmt.Errorf("source file chain id %s and current node chain id %s is different", networkInfo.ChainID, ctx.ChainID())
+		}
```
Comment (pbukva) [inline app/app.go:743]:
```suggestion
		if networkInfo.MergeSourceChainID != mergeSourceGenesisDoc.ChainID {
			return nil, fmt.Errorf("source file chain id %s and current node chain id %s is different", networkInfo.ChainID, ctx.ChainID())
		}
		if networkInfo.DestinationChainID != ctx.ChainID() {
			return nil, fmt.Errorf("source file chain id %s and current node chain id %s is different", networkInfo.ChainID, ctx.ChainID())
		}
```
See my [previous post](https://github.com/fetchai/fetchd/pull/383/files/959f5dd25a6f9ef8fd5a512405ee79c400879fe0..615a29b4b604b9aecbbff8ceae440357058b5251#r1787894379) .

File: app/upgrade_cudos_distribution.go (hunk 5 above)
Comment (MissingNO57) [reply app/upgrade_cudos_distribution.go:404]: cudos merge prefix is added to error in `CudosMergeUpgradeHandler`

**Review by pbukva — APPROVED** [review]
LGTM

**Pull request merged**